diff --git a/modules/dashboard/dash/User.qml b/modules/dashboard/dash/User.qml
index bf989c6..b7bbe55 100644
--- a/modules/dashboard/dash/User.qml
+++ b/modules/dashboard/dash/User.qml
@@ -69,7 +69,7 @@ Row {
                 property string uptime
 
                 running: true
-                command: ["uptime", "-p"]
+                command: ["sh", "-c", `uptime | awk -F, '{sub(".*up ",x,$1);print $1}' | sed -e 's/^[ \\t]*//'`]
                 stdout: StdioCollector {
                     onStreamFinished: uptimeProc.uptime = text.trim()
                 }
